### DISCUSSION TITLE
macro and shell customization menu location

### DISCUSSION THREAD

**Discussion by tksoh** [eteran/nedit-ng#232] [Ideas]
NEdit keep the menus to bring up the dialog for customizing shell and macros three levels deep under the Preferences menu. I feel it would be more intuitive to have them under the Shell and Macro menus, not to mention it's much quicker to access.

Now that NG maintains separate YAML files for macro and shell, we could just prompt to save them when we exit the dialogs.

**Comment by eteran** [comment]
I was actually thinking **just this afternoon** that the Macro and Shell "edit" menus could be at the top of each of them! (great minds think alike, LOL).

I like the idea and it's trivial to implement.

And you're right, we could absolutely just prompt to save when closing that dialog.

This would also play nicely with the "unified preferences dialog" plan since those are the more complex parts. If we treat the macro/shell dialogs as their own thing outside of the main preferences dialog, that's a LOT less work to do :-).

**Comment by grege2** [comment]
Agree very much about moving Macro menu higher in the hierarchy. When you're customising macros and key bindings etc, esp. in the initial transition from Nexit-X11 to Nedit-ng,  you're **constantly** diving down several layers in the menus.

**Comment by tksoh** [comment]
For the sake of transition from nedit 5->6, perhaps we should  **_add_** the new menus, rather than **_move_**?

**Comment by eteran** [comment]
Not a bad idea, I'm not the biggest fan of redundancy, but in a situation like this, it may be justified, at least for one release cycle.
